### PR TITLE
Feature/e2e tests for create list

### PR DESCRIPTION
**Pull request by miv1**
[Trello](https://trello.com/c/H06dXxZV/7-e2e-tests-for-create-a-list-project)

**Overview**
- Adding scenarios about the creation of list.
- Adding all necessary steps of the scenarios.
- Modifying the config for run one by one the features.

**Dependencies**
None

### CODE REVIEW

**Review by R0m3rCh — APPROVED** [review]

**Review by robertocgongora — DISMISSED** [review]

File: test/e2e/steps/hooks.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ Before('@loginHook', function () {
 After('@deleteList', function () {
   DashboardPage.listSettingsButton.click();
   DashboardPage.deleteButton.click();
-  DashboardPage.confirmDeleteButton.waitForExist(3000);
+  //Wait until button is enable
+  browser.pause(2000);
```
Comment (robertocgongora) [inline test/e2e/steps/hooks.js]: find another way to wait for an element to be available

File: test/e2e/steps/list.steps.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  browser.keys(SystemInteractions.ENTER_KEY_PRESS);
+  browser.keys(SystemInteractions.ESCAPE_KEY_PRESS);  
+  BoardPage.closePanelTask.click();
+});
+
+When(/^the user creates a new folder "([^"]*)"$/, function (nameList) {
+  BoardPage.templateLink.waitForExist(3000);
+  BoardPage.templateLink.click();  
+  //Wait for 'New' button
+  browser.pause(3000);
```
Comment (robertocgongora) [inline test/e2e/steps/list.steps.js]: find another way to wait for an element to be available

File: wdio.conf.js [hunk 3]
```diff
@@ -49,7 +49,7 @@ exports.config = {
   // and 30 processes will get spawned. The property handles how many capabilities
   // from the same test should run tests.
   //
-  maxInstances: 10,
+  maxInstances: 1,
```
Comment (robertocgongora) [inline wdio.conf.js:52]: is this to execute just one instance? if so, please add this new feature in your PR's description. So the code reviewer will be aware of this change

**Review by robertocgongora — COMMENTED** [review]
Please, fix the observations before merging

**Review by robertocgongora — APPROVED** [review]

File: test/e2e/features/list.feature [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  Scenario: The user creates a list from creation of tasks
+    Given the user is at Dashboard page
+    When the user clicks on New task button
+    And the user creates a new list "List from task options"
+    Then "List from task options" list should be created
+
+  Scenario: The user creates a list from creation of a folder
+    Given the user is at Dashboard page
+    When the user creates a new folder "List within folder"
+    Then "List" list should be created within the folder
```
Comment (lolpez) [inline test/e2e/features/list.feature]: Maybe
`Then "List" list should be created within the folder "List within folder"`

File: test/e2e/features/login.feature [hunk 5]
```diff
@@ -4,8 +4,8 @@ Feature: Login
   Scenario: User Logs to ClickUp with valid credentials
     Given user navigates to "/login"
     When user fills the fields with the following information
-    |mail                   |password       |
-    |nexefeb197@4tmail.com  |1234Qwert      |
-    |invalid@gmail.com      |demoPassword   |
+      | mail                  | password     |
+      | nexefeb197@4tmail.com | 1234Qwert    |
+      | invalid@gmail.com     | demoPassword |
     Then user should be redirected to ClickUp's home page
```
Comment (lolpez) [inline test/e2e/features/login.feature]: Maybe
Then user should be redirected to ClickUp's dashboard
Home Page might refer to the actual index page of Clickup

File: test/e2e/pages/BoardPage.js [hunk 6]
```diff
@@ -1,26 +1,53 @@
...
+
+  get searchList() { return $('.cu-dropdown__toggle-text'); }
+
+  get newListLink() { return $('.category-list__add-category-link'); }
+
+  get listBySearchInput() { return $('.category-list-input__input'); }
+
+  get listNameInSpace() { return $('.cu-nav-section'); }
+
+  get listTab() { return $('div=List'); }
```
Comment (lolpez) [inline test/e2e/pages/BoardPage.js:28]: I am not sure what this locatior does.  Can you explain it or show an example?

**Review by lolpez — COMMENTED** [review]
Added 3 comments

File: test/e2e/features/login.feature (hunk 5 above)
Comment (miv1) [reply test/e2e/features/login.feature]: I didn't touch this in my task but is done

File: test/e2e/pages/BoardPage.js (hunk 6 above)
Comment (miv1) [reply test/e2e/pages/BoardPage.js:28]: sure, it is for get in to the list tab where the tasks and the status are.

File: test/e2e/features/list.feature [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  Scenario: The user creates a list from creation of tasks
+    Given the user is at Dashboard page
+    When the user clicks on New task button
+    And the user creates a new list "List from task options"
+    Then "List from task options" list should be created
+
+  Scenario: The user creates a list from creation of a folder
+    Given the user is at Dashboard page
+    When the user creates a new folder "List-within-folder"
+    Then "List" list should be created within the folder List-within-folder
```
Comment (lolpez) [inline test/e2e/features/list.feature:21]: you forgot the quotes "List-within-folder"

Comment (miv1) [reply test/e2e/features/list.feature:21]: I am not sending this name as parameter, the creation of "List" is being compared.

Comment (lolpez) [reply test/e2e/features/list.feature:21]: got it

**Review by lolpez — APPROVED** [review]

**Review by lolpez — APPROVED** [review]

**Pull request merged**